DataBindings/Botify
# Spotify search song

**Description**:

This PR introduces the `searchSong` function, an asynchronous function that queries songs on the Spotify platform using the Spotify Web API. It also encompasses a set of unit tests to validate the behavior of the function across a spectrum of situations.

**Changes Made**:

- Added the `searchSong` function to perform song searches via the Spotify Web API.
- Incorporated comprehensive error handling for multiple response status codes, including scenarios such as incorrect bearer authentication, token expiration, bad OAuth requests, rate limit exceedance, server errors, and unknown error situations.
- Implemented unit tests using Jest and fetchMock to ensure the `searchSong` function operates correctly in various contexts.
- Documented the code and introduced comments to enhance code clarity and align with project standards.

**Motivation and Context**:

This PR augments the project by introducing a versatile function for accessing Spotify songs, streamlining development for applications that require this functionality. Furthermore, it establishes dependable error handling and consistent data retrieval.

**How to Test**:

1. Ensure the presence of the necessary environment variables: `SPOTIFY_CLIENT_ID`, `SPOTIFY_CLIENT_SECRET`, and `SPOTIFY_REFRESH_TOKEN` must be correctly configured.

2. Run the included unit tests, leveraging a testing framework such as Jest, to confirm the correctness of the `searchSong` function across diverse scenarios.

**Checklist**:

- [x] Added the `searchSong` function.
- [x] Implemented robust error handling for a range of response status codes.
- [x] Developed unit tests covering diverse scenarios.
- [x] Integrated fetchMock for reliable testing.
- [x] Documented code and included comments for enhanced code clarity.
- [x] Code conforms to project coding standards.

**Related Issues**:

N/A
